core/trino-main/src/main/java/io/trino/FeaturesConfig.java

```diff
@@ -491,4 +497,17 @@ public FeaturesConfig setForceSpillingJoin(boolean forceSpillingJoin)
         this.forceSpillingJoin = forceSpillingJoin;
         return this;
     }
+
+    public int getJoinPositionsArrayCutoff()
+    {
+        return joinPositionsArrayCutoff;
+    }
+
+    @Config("join-positions-array-cutoff")
```

Comment:
Instead of this flag, which looks hard to understand and tune for users, can we estimate the potential memory usage of each PagesHash implementation and pick the one with lesser memory requirement in JoinHashSupplier ?

Reply:
The new implementation is always (almost) using more memory. So this is just to prevent unlimited increase